eq-author-api/src/validation/customKeywords/validateRoutingLogicalAND.js

```diff
@@ -0,0 +1,137 @@
+const { groupBy } = require("lodash");
+const { getAnswerById } = require("../../../schema/resolvers/utils");
+const {
+  CURRENCY,
+  NUMBER,
+  PERCENTAGE,
+  UNIT,
+} = require("../../../constants/answerTypes");
+const createValidationError = require("../createValidationError");
+const { ERR_LOGICAL_AND } = require("../../../constants/validationErrorCodes");
+
+const getRHS = e =>
+  e.right && e.right.customValue && e.right.customValue.number;
+
+const areUnanswered = e => e.condition === "Unanswered";
+
+module.exports = ajv => {
+  ajv.addKeyword("validateRoutingLogicalAND", {
+    validate: function isValid(
+      otherFields,
+      { expressions },
+      fieldValue,
+      dataPath,
+      parentData,
+      fieldName,
+      questionnaire
+    ) {
+      const invalidAnswerIds = new Set();
+      const expressionsByAnswerId = groupBy(expressions, "left.answerId");
+      const potentialConflicts = Object.entries(expressionsByAnswerId).filter(
+        ([answerId, expressions]) =>
+          expressions.length > 1 && answerId && answerId.length
+      );
+
+      const error = answerId => invalidAnswerIds.add(answerId);
+
+      potentialConflicts.forEach(([answerId, expressions]) => {
+        // Handle invalid combination of "Unanswered" with any answer-requiring condition
+        if (
+          expressions.some(areUnanswered) &&
+          !expressions.every(areUnanswered)
+        ) {
+          return error(answerId);
+        }
```

Comment:
Yeah, it does already mean the same thing. I'm kind of attached to having it as a function just because I think it seems neater and less error-prone than calling `return invalidAnswerIds.add()` several times - how about I rename it to `registerError()` or something instead? ;-)

I get what you mean though. It's only an alias for `invalidAnswerIds.add`. Can change if you want.

Reply:
I've renamed it to `addError`. Let me know if you disagree ;-)

Reply:
It's okay, I'm not emotionally attached and I can get behind the 'it looks neater' argument. 😉 The new name you've given it is better. Thanks, Tom.